documentation/SolutionsRegistryFormat.md

```diff
@@ -143,6 +143,37 @@ This directive is used to detect whether a solution is installed. If any of thes
 ]
 ```
 
+####isConfigurable
+This is run before configuration to ensure that the application is actually ready to be configured. This is relevant for applications where e.g. a configuration file needs to be present, a wizard needs to be run on the first launch, etc. Each of the blocks will be evaluated, if *any* of them evaluates to true, the solution is considered configurable. The absence of an `isConfigurable` block is also interpreted as the solution being configurable.
+
+If a solution is not configurable, the `makeConfigurable` block will be executed immediately following the `isConfigurable` check (see below).
+
+**Example Entry**:
+```
+"isConfigurable": [
+    {
+        "type": "gpii.lifecycleActions.pathExists",
+        "path": "/tmp/fakemag1.settings.json"
+    }
+]
+```
+
+####makeConfigurable
+Is the actions that need to be taken to make the application configurable (such as running a wizard, creating a default configuration file, adding a new system user, etc). This will be triggered by the system if `isConfigurable` has evaluated to false and run immediately following the `isConfigurable` check.
+
+**Example Entry**:
+```
+"makeConfigurable": [
+    {
+        "type": "gpii.lifecycleActions.createFile",
```

Comment:
Both of these examples now refer to nonexistent functions. I think it would be better to reinstate them, with some tests,  so that we create a facility which is actually usable. Don't we have an actual use case for this feature - if not, why are we implementing it? Unfortunately GPII-1885 is a TLJ so the reason why it is on our roadmap is unclear.

Could you please
i) Flesh out GPII-1885 by adding or linking to information explaining why we are implementing it
ii) Write up a JIRA for the facility of being able to invoke settingsHandlers from the makeConfigurable/isConfigurable blocks together with a sketch of what this might look like